plugins/xdebug/xdebug_headers.cc

```diff
@@ -191,14 +222,70 @@ print_response_headers(TSHttpTxn txn, std::stringstream &output)
   TSMLoc    hdr_loc;
   if (TSHttpTxnServerRespGet(txn, &buf_s, &hdr_loc) == TS_SUCCESS) {
     output << "{'type':'response', 'side':'server', 'headers': {\n";
-    print_headers(buf_s, hdr_loc, output);
+    print_headers(buf_s, hdr_loc, output, JSON_COMPLIANT);
     output << "\n\t}},";
     TSHandleMLocRelease(buf_s, TS_NULL_MLOC, hdr_loc);
   }
   if (TSHttpTxnClientRespGet(txn, &buf_c, &hdr_loc) == TS_SUCCESS) {
     output << "{'type':'response', 'side':'client', 'headers': {\n";
-    print_headers(buf_c, hdr_loc, output);
+    print_headers(buf_c, hdr_loc, output, JSON_COMPLIANT);
     output << "\n\t}}";
     TSHandleMLocRelease(buf_c, TS_NULL_MLOC, hdr_loc);
   }
 }
+
+void
+print_request_headers_json(TSHttpTxn txn, std::stringstream &output)
+{
+  TSMBuffer buf_c, buf_s;
+  TSMLoc    hdr_loc;
+
+  bool has_client = false;
+
+  Dbg(dbg_ctl_hdrs, "Printing client request headers for full JSON");
+  if (TSHttpTxnClientReqGet(txn, &buf_c, &hdr_loc) == TS_SUCCESS) {
+    output << "{\"client-request\":{";
+    print_headers(buf_c, hdr_loc, output, !JSON_COMPLIANT);
```

Comment:
What does the `_json` version of print request headers pass `false` for the `json_compliant` flag?